### PR TITLE
[#165] Keycloak support added

### PR DESCRIPTION
**Pull request by filip-kopecky**
Added keycloak

### CODE REVIEW

**Comment by filip-kopecky** [comment]
Added profiles to the relevant test classes + removed isAuthenticated from `/configuration` endpoint. The previously failed tests should be working now. @ledsoft

File: pom.xml [hunk 1]
```diff
@@ -454,6 +482,8 @@
                             <version>${org.hibernate.validator.version}</version>
                         </path>
                     </annotationProcessorPaths>
+                    <source>11</source>
+                    <target>11</target>
```
Comment (ledsoft) [inline pom.xml]: Replace with `java.version` from properties.

File: src/main/java/cz/cvut/kbss/termit/config/SecurityKeycloakConfig.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    }
+
+    @Autowired
+    public void configureGlobal(
+            AuthenticationManagerBuilder auth) throws Exception {
+
+        KeycloakAuthenticationProvider keycloakAuthenticationProvider
+                = keycloakAuthenticationProvider();
+        keycloakAuthenticationProvider.setGrantedAuthoritiesMapper(
+                new SimpleAuthorityMapper());
```
Comment (ledsoft) [inline src/main/java/cz/cvut/kbss/termit/config/SecurityKeycloakConfig.java]: Replace with a custom authority mapper looking like this:
```java
public class HierarchicalRoleBasedAuthorityMapper implements GrantedAuthoritiesMapper {

    @Override
    public Collection<? extends GrantedAuthority> mapAuthorities(Collection<? extends GrantedAuthority> authorities) {
        return authorities.stream().map(a -> {
                              final UserRole role = UserRole.fromRoleName(a.getAuthority());
                              return role.getGranted();
                          }).flatMap(roles -> roles.stream().map(r -> new SimpleGrantedAuthority(r.getName())))
                          .collect(Collectors.toSet());
    }
}
```

**Review by ledsoft — CHANGES_REQUESTED** [review]
I figured out the problem with `PreAuthorize` - TermIt internal roles work hierarchically, so `ROLE_FULL_USER` automatically includes `ROLE_RESTRICTED_USER`. Keycloak-based authorization does not work like this.
To fix it, create a custom authority mapper and pass it to `KeycloakAuthenticationProvider`. See the inline comment. After that, all the extra `PreAuthorize("isAuthenticated()")` can be removed and the PR will me much more compact.

**Review by ledsoft — APPROVED** [review]
LGTM

**Pull request merged**